lib/internal/modules/cjs/loader.js

```diff
@@ -789,36 +800,42 @@ function wrapSafe(filename, content) {
       } : undefined,
     });
   }
+  try {
+    const compiled = compileFunction(
+      content,
+      filename,
+      0,
+      0,
+      undefined,
+      false,
+      undefined,
+      [],
+      [
+        'exports',
+        'require',
+        'module',
+        '__filename',
+        '__dirname',
+      ]
+    );
```

Comment:
The try statement must end its wrap here, otherwise we risk intercepting other errors as well, which we shouldn't.

Reply:
will work by defining with `let` and fn outside the try-catch it the variable is visible to where its used.

Reply:
should be fine now